consensus/istanbul/backend/backend.go

```diff
@@ -780,6 +795,9 @@ func (sb *Backend) addProxy(node, externalNode *enode.Node) error {
 	sb.p2pserver.AddPeer(node, p2p.ProxyPurpose)
 
 	sb.proxyNode = &proxyInfo{node: node, externalNode: externalNode}
+	if err := sb.updateAnnounceVersion(newAnnounceVersion()); err != nil {
```

Comment:
Does this need to be done here?  Seems like proxy is send the enodeCertificate in `addPeer`.

Reply:
I like having this here so that the announce version is definitely updated whenever there's a change in proxies (I know the RPC isn't well tested right now but better to support it)

But you're right, if `sb.p2pserver.AddPeer(node, p2p.ProxyPurpose)` (which is not synchronous) causes the peer to be registered before `updateAnnounceVersion` is called, the peer would have been sent the old enodeCertificate when it was registered, and then immediately after a new one when `updateAnnounceVersion` is called. I'm going to move `updateAnnounceVersion` to be called before `AddPeer` so that the proxy is guaranteed to only be sent the correct message when it's registered as a peer.